tests/mpi/periodicity_03.cc

```diff
@@ -143,11 +143,10 @@ namespace Step22
    const TrilinosWrappers::MPI::Vector &src) const
   {
     SolverControl solver_control (10*src.size(), 1e-6*src.l2_norm(), false, false);
-    TrilinosWrappers::SolverCG    cg (solver_control,
-                                      TrilinosWrappers::SolverCG::AdditionalData());
+    SolverGMRES<TrilinosWrappers::MPI::Vector> gmres(solver_control);
 
     tmp = 0.;
-    cg.solve (*matrix, tmp, src, *preconditioner);
+    gmres.solve (*matrix, tmp, src, *preconditioner);
```

Comment:
I still don't think I understand this. You mentioned that you had convergence problems. But why? This solver is applied to the top left block of the matrix, which should be symmetric and positive definite. Is it indefinite (i.e., has a null space) because you have periodic boundary conditions? I'd like to understand where the problem is coming from before merging this.